### PR TITLE
Define AWS Secrets Store provider

### PR DESCRIPTION
**Pull request by sgrebnov**
Adds new provider to retrieve secrets from AWS Secrets Manager. Could be enabled via
```
secrets:
  store: aws_secrets_manager
```

Implementation for `AwsSecretStore` provider will be added separately

### CODE REVIEW

File: crates/spicepod/src/component/secrets.rs [hunk 1]
```diff
@@ -43,4 +43,5 @@ pub enum SpiceSecretStore {
     Env,
     Kubernetes,
     Keyring,
+    Aws,
```
Comment (phillipleblanc) [inline crates/spicepod/src/component/secrets.rs]:
```suggestion
    AwsSecretStore,
```

File: crates/secrets/src/lib.rs [hunk 2]
```diff
@@ -80,17 +82,21 @@ pub enum SecretStoreType {
     #[cfg(feature = "keyring-secret-store")]
     Keyring,
     Kubernetes,
+    #[cfg(feature = "aws-secret-store")]
+    AwsSecretStore,
 }
 
 #[must_use]
+#[allow(unreachable_patterns)]
```
Comment (digadeesh) [inline crates/secrets/src/lib.rs]: I don't think this is right.   You removed the last option so I think its an unreachable pattern now.  I'd rather not suppress these types of warnings unless we really have to.

File: crates/spicepod/src/component/secrets.rs (hunk 1 above)
Comment (phillipleblanc) [reply crates/spicepod/src/component/secrets.rs]: Probably needs to be serialized as snake case as well

File: crates/secrets/src/lib.rs [hunk 3]
```diff
@@ -14,6 +14,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+#[cfg(feature = "aws-secret-store")]
+pub mod aws;
```
Comment (lukekim) [inline crates/secrets/src/lib.rs]: Let's make this specific. `aws_secrets_manager`

File: crates/secrets/src/lib.rs (hunk 2 above)
Comment (sgrebnov) [reply crates/secrets/src/lib.rs]: Updated, thank you 
```
#[cfg(not(all(feature = "keyring-secret-store", feature = "aws-secrets-manager")))]
 _ => None,
```

File: crates/spicepod/src/component/secrets.rs (hunk 1 above)
Comment (sgrebnov) [reply crates/spicepod/src/component/secrets.rs]: Updated. Thank you @phillipleblanc !
```
    #[serde(rename = "aws_secrets_manager")]
    AwsSecretsManager,
```

File: crates/secrets/src/lib.rs (hunk 3 above)
Comment (sgrebnov) [reply crates/secrets/src/lib.rs]: Updated to `aws_secrets_manager` (file name, struct name, feature flag so we have consistent naming everywhere). Thank you @lukekim!

**Review by digadeesh — APPROVED** [review]

**Pull request merged**